common/scala/src/main/scala/whisk/http/PoolingRestClient.scala

```diff
@@ -81,14 +91,14 @@ class PoolingRestClient(protocol: String, host: String, port: Int, queueSize: In
     }
   }
 
-  protected def mkRequest(method: HttpMethod, uri: Uri, headers: List[HttpHeader] = List.empty): Future[HttpRequest] = {
+  def mkRequest(method: HttpMethod, uri: Uri, headers: List[HttpHeader] = List.empty): Future[HttpRequest] = {
     mkRequest0(method, uri, Future.successful(HttpEntity.Empty), headers)
   }
```

Comment:
If we make all of those public, would it make sense to pull them out in a `RestClient` object? They don't need the context of a class and are merely small plumbing helpers.